### PR TITLE
monorepo

### PR DESCRIPTION
**Pull request by vimwitch**
Makes a monorepo with `backend`, `frontend`, and `unirep-social` (as core). To give it a try use the following commands:

```
git clone -b monorepo git@github.com:Unirep/Unirep-Social.git unirep-social
cd unirep-social
yarn && yarn bootstrap --force-local
```

Now the repo should be functional.

```
cd packages/core
yarn build && yarn test
```

```
cd packages/backend
yarn test
```

```
cd packages/frontend
yarn start
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,57 +1,17 @@
 {
-    "name": "unirep-social",
-    "version": "1.0.0",
-    "description": "",
-    "main": "build/core/index.js",
+    "name": "root",
```
Comment (vivianjeng) [inline package.json:2]: I think it can be named`unirep-social`

Comment (vimwitch) [reply package.json:2]: The monorepo package shouldn't be named anything, it's not a valid package.

File: package.json [hunk 2]
```diff
@@ -1,57 +1,17 @@
...
-        "url": "git+https://github.com/Unirep/Unirep-Social.git"
-    },
-    "author": "",
-    "license": "MIT",
-    "bugs": {
-        "url": "https://github.com/Unirep/Unirep-Social/issues"
-    },
-    "homepage": "https://github.com/Unirep/Unirep-Social#readme",
+    "private": true,
     "devDependencies": {
```
[the remark this answers is not in the document]
Comment (vimwitch) [reply package.json:10]: I don't understand, what do we need to do?

File: packages/backend/LICENSE [hunk 3]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
[the remark this answers is not in the document]
Comment (vimwitch) [reply packages/backend/LICENSE:1]: Yes, they should already be in the root directory, just forgot to remove the backend one.

**Comment by vimwitch** [comment]
@vivianjeng Remember, we're not merging this, we're changing the default branch to `main`. I also can't archive those repos.

**Comment by kittybest** [comment]
Inside `packages/core/cli/test/testAllCommands.ts`
change `import { UnirepSocialFactory } from '../../core/utils'` to `import { UnirepSocialFactory } from '../../src/utils'`
this will cause the `yarn test-cli` failed

**Comment by vimwitch** [comment]
Ah thanks, didn't realize the cli had tests

**Comment by vimwitch** [comment]
Merged into `main`

**Pull request closed without merging**